### PR TITLE
Row height reaction

### PR DESCRIPTION
**Pull request by peterjcoffman**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- N/A Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.
https://github.com/xh/toolbox/pull/331

Another Toolbox branch:
https://github.com/xh/toolbox/tree/configurableItemHeight
We won't want to merge this Toolbox branch; it's just for testing.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by TomTirapani** [comment]
I see you've added Lee as a reviewer - would like to get his input on adding another (albeit minor) breaking change to this release before you go ahead and make the change suggested above

**Comment by lbwexler** [comment]
@TomTirapani -- think you suggested change is spot on, and apologies peter for not pointing you in that direction earlier  (peter had asked about). Its definitely consistent with Hoist conventions to have the dynamic API on any model present, and putting this side by side with itemRender makes its a clear winner.

Peter -- you will get to use your new apiRemoved tool!  If you get this one in first thing, can still make it for Hoist 29.0.0.

**Comment by peterjcoffman** [comment]
> Works great!
> 
> I think we could simplify this significantly by making `itemHeight` something you specify on the `DataViewModel` rather than the `DataView` component.

I agree. I was trying to avoid a breaking change, but that is simpler.

itemHeight is currently required and we're making it forbidden, so this will break _all_ dataViews.

**Comment by peterjcoffman** [comment]
I opened a new Toolbox PR that we will actually want to merge:
https://github.com/xh/toolbox/pull/331

File: cmp/dataview/DataViewModel.js [hunk 1]
```diff
@@ -76,6 +83,11 @@ export class DataViewModel {
             ]
         });
 
+        this.setItemHeight(itemHeight);
```
Comment (lbwexler) [inline cmp/dataview/DataViewModel.js]: Don't need to use the setter here in the constructor.

File: cmp/dataview/DataViewModel.js [hunk 2]
```diff
@@ -76,6 +83,11 @@ export class DataViewModel {
...
+        this.addReaction({
+            track: () => this.itemHeight,
+            run: () => this.gridModel?.agApi.resetRowHeights()
```
Comment (lbwexler) [inline cmp/dataview/DataViewModel.js]: Its actually the agApi that can be null here

File: cmp/dataview/DataViewModel.js [hunk 3]
```diff
@@ -36,17 +40,20 @@ export class DataViewModel {
...
         sortBy = [],
         selModel,
         emptyText,
-        contextMenu = null
+        contextMenu = null,
+        itemHeight
     }) {
         sortBy = castArray(sortBy);
         throwIf(sortBy.length > 1, 'DataViewModel does not support multiple sorters.');
+        throwIf(!isNumber(itemHeight), 'Must specify a number for itemHeight in DataView model.');
```
Comment (TomTirapani) [inline cmp/dataview/DataViewModel.js]: MInor nit, but 'DataView model'  should be one word: 'DataViewModel'

File: cmp/dataview/DataViewModel.js [hunk 4]
```diff
@@ -36,17 +40,20 @@ export class DataViewModel {
      * @param {(Object[]|GridStoreContextMenuFn)} [c.contextMenu] - array of RecordActions, configs or token
      *      strings with which to create grid context menu items.  May also be specified as a
      *      function returning a StoreContextMenu.  Desktop only.
+     * @param {number} itemHeight - Row height for each item displayed in the view
```
Comment (TomTirapani) [inline cmp/dataview/DataViewModel.js]: Can be order this in the comment and constructor params below to come after the itemRenderer? We usually put required params first, before the optional ones

**Pull request merged**